### ISSUE TITLE
db-batch gets stuck in a loop

### ISSUE THREAD

**Issue by garethbowen** [medic/medic#5319]
Run the `extract-read-status` migration over a database with plenty of data. You can force the migration to run by manually removing it from the array in the `migration-log` document and restarting api.

What should happen: it runs through to completion then api starts.
What happens: it gets stuck in a loop and never completes and api never starts up

Caused by https://github.com/medic/medic/commit/a263801d50e36e446ca8bf9b5961611d9d9ad3e8

**Comment by garethbowen** [comment]
It turns out PouchDB doesn't support pagination using `startkey` and `startkey_docid`.

**Comment by garethbowen** [comment]
Merged and backported to `3.4.x`

**Comment by newtewt** [comment]
I'm trying to reproduce this in 3.3.x and not seeing it hang. I assume I don't have enough docs.  Was hoping to do a before and after check to call this good. 


```
2019-02-13 12:00:23 INFO: Running db migrations… 
2019-02-13 12:00:23 INFO: Running migration extract-read-status... 
2019-02-13 12:00:23 INFO:         Processing doc 86 
2019-02-13 12:00:23 INFO:         Processing doc 186 
2019-02-13 12:00:23 INFO:         Processing doc 286 
2019-02-13 12:00:23 INFO:         Processing doc 386 
2019-02-13 12:00:23 INFO:         Processing doc 486 
2019-02-13 12:00:23 INFO:         Processing doc 586 
2019-02-13 12:00:23 INFO:         Processing doc 686 
2019-02-13 12:00:23 INFO:         Processing doc 786 
2019-02-13 12:00:23 INFO:         Processing doc 886 
2019-02-13 12:00:23 INFO:         Processing doc 986 
2019-02-13 12:00:23 INFO:         Processing doc 1086 
2019-02-13 12:00:23 INFO:         Processing doc 1186 
2019-02-13 12:00:23 INFO:         Processing doc 1286 
2019-02-13 12:00:23 INFO:         Processing doc 1386 
2019-02-13 12:00:23 INFO:         Processing doc 1486 
2019-02-13 12:00:23 INFO: Migration extract-read-status completed successfully 
2019-02-13 12:00:23 INFO: Database migrations completed successfully 
2019-02-13 12:00:23 INFO: Medic API listening on port 5988 

```

**Comment by garethbowen** [comment]
@newtewt I think the bug was introduced in 3.4.x and fixed in 3.4.x so you won't be able to reproduce it in 3.3.x. What I was seeing is it saying "processing doc x" with the same value for x over and over again, so what you're seeing proves it works. If you can run this same test on 3.4.x and get the same output I think we can close this.

**Comment by newtewt** [comment]
Awesome, I have upgrade the my local instance to 3.4.x and seeing it process as above.
